src/components/timeline/GlobalTrack.js

```diff
@@ -0,0 +1,279 @@
+    }
+  };
+
+  renderTrack() {
+    const { globalTrack } = this.props;
+    switch (globalTrack.type) {
+      case 'process': {
+        const { mainThreadIndex } = globalTrack;
+        if (mainThreadIndex === null) {
+          return <div className="timelineTrackThreadBlank" />;
+        }
+        return <TimelineTrackThread threadIndex={mainThreadIndex} />;
+      }
+      case 'screenshots':
+        // TODO: Add support for screenshots.
+        return <div />;
+      default:
+        console.error('Unhandled globalTrack type', (globalTrack: empty));
+        return null;
+    }
+  }
+
+  _changeLocalTrackOrder = (trackOrder: TrackIndex[]) => {
+    const { globalTrack, changeLocalTrackOrder } = this.props;
+    if (globalTrack.type === 'process') {
+      // Only process tracks have local tracks.
+      changeLocalTrackOrder(globalTrack.pid, trackOrder);
+    }
+  };
+
+  render() {
+    const {
+      isSelected,
+      isHidden,
+      titleText,
+      trackName,
+      style,
+      localTracks,
+      localTrackOrder,
+      pid,
+    } = this.props;
+
+    if (isHidden) {
+      // If this global track is hidden, render out a stub element so that the
+      // Reorderable Component still works across all the tracks.
+      return <li className="timelineTrackHidden" />;
+    }
+
+    return (
+      <li className="timelineTrack" style={style}>
+        <div
+          className={classNames('timelineTrackRow timelineTrackGlobalRow', {
+            selected: isSelected,
+          })}
+          onClick={this._onLineClick}
+        >
+          <ContextMenuTrigger
+            id={'TimelineTrackContextMenu'}
+            renderTag="div"
+            attributes={{
+              title: titleText,
+              className: 'timelineTrackLabel timelineTrackGlobalGrippy',
+              onMouseDown: this._onLabelMouseDown,
+            }}
+          >
+            <h1 className="timelineTrackName">{trackName}</h1>
+          </ContextMenuTrigger>
+          <div className="timelineTrackTrack">{this.renderTrack()}</div>
+        </div>
+        <Reorderable
+          tagName="ol"
+          className="timelineTrackLocalTracks"
+          order={localTrackOrder}
+          orient="vertical"
+          grippyClassName="timelineTrackLocalGrippy"
+          onChangeOrder={this._changeLocalTrackOrder}
+        >
+          {localTracks.map((localTrack, trackIndex) => {
+            if (pid === null) {
+              console.error(
+                'The pid should never be null when adding a TimelineLocalTrack.',
+                pid
+              );
+              return <div />;
+            }
+            return (
+              <TimelineLocalTrack
+                key={trackIndex}
+                pid={pid}
+                localTrack={localTrack}
+                trackIndex={trackIndex}
+              />
+            );
+          })}
+        </Reorderable>
+      </li>
+    );
+  }
+}
+
+function getGlobalTrackName(
```

Comment:
What's the difference with the one in `profile-logic/tracks` ? They look subtly different but the other one seems more complete (works with a string pid especially).

Reply:
This was some mistaken code duplication. I removed it and centralized my usage of getting track names around 2 selectors, one for global track names, and the other for local ones.